### ISSUE TITLE
--from-file positional parameter not passed to jq since 2.3.5

### ISSUE THREAD

**Issue by brydoncheyney** [kislyuk/yq#19]
After changes made to passing positional parameters to jq, it looks like `--from-file filter` is broken and the filter for jq __must__ be defined as the `jq_filter` of `yq`. This means you can no longer define filters in a file, which is invaluable for more complex constructs.

## Example Setup

Define a filter and some yaml data

``` console
[yq] cat filter
.foo
[yq] cat data
foo
  a: 1
```

## 2.3.4

Positional arguments are broken

```console
[yq] yq -y --arg a b . data
usage: yq [-h] [--yaml-output] [--width WIDTH] [--version]
          jq_filter [files [files ...]]
yq: error: argument files: can't open 'b': [Errno 2] No such file or directory: 'b'
```
`--from-file` without `jq_filter` works

```console
[yq] yq -y --from-file filter data
a: 1
```

## 2.3.5

After changes to positional parameters, the `--args` example now works

```console
[yq] yq -y --arg a b . data
foo:
  a: 1
```

...but the `--from-file` example is broken, returning the `--help` text with a return code 0.

```console
[yq] yq -y --from-file filter data
usage: yq [options] <jq filter> [YAML file...]
          jq_filter [files [files ...]]

yq: Command-line YAML processor - jq wrapper for YAML documents
...
[yq] echo $?
0
```

trying to define the call with the self identify filter still fails to work, returning no error message but a return code of 2.

```console
[yq] yq -y --from-file filter . data
[yq] echo $?
2
```

**Comment by brydoncheyney** [comment]
Updated the issue after further investigation revealed the initial report was over simplified.

**Comment by kislyuk** [comment]
Thank you for investigating. Looking into this.

**Comment by kislyuk** [comment]
This should be fixed in v2.3.7. Please test.
